internal/stacksapi/client.go

```diff
@@ -0,0 +1,261 @@
+package stacksapi
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log/slog"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/buildkite/agent-stack-k8s/v2/internal/version"
+	"github.com/buildkite/roko"
+)
+
+type Client struct {
+	baseURL    *url.URL
+	userAgent  string
+	authHeader string
+
+	logger      *slog.Logger
+	httpClient  *http.Client
+	retrierOpts []roko.RetrierOpt
+}
+
+func urlMustParse(u string) *url.URL {
+	parsed, err := url.Parse(u)
+	if err != nil {
+		panic(err)
+	}
+	return parsed
+}
+
+type ClientOpt func(*Client) error
+
+const (
+	DefaultBaseURL = "https://agent.buildkite.com/v3/"
+	DefaultQueue   = "_default"
+)
+
+var (
+	DefaultRetrierOptions = []roko.RetrierOpt{
+		roko.WithMaxAttempts(5),
+		roko.WithStrategy(roko.Constant(1 * time.Second)),
+		roko.WithJitterRange(500*time.Millisecond, 1*time.Second),
+	}
```

Comment:
i have no idea if this is a good default retry policy, i basically just put something on paper and went with it. happy to do something better.

Reply:
For various reasons, I would make the default strategy `ExponentialSubsecond`.